sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -710,7 +710,7 @@ public static Row toBeamRow(Schema rowSchema, TableSchema bqSchema, TableRow jso
                       (!innerTypeIsMap && v instanceof Map)
                           ? ((Map<String, Object>) v).get("v")
                           : v)
-              .map(v -> toBeamValue(fieldType.getCollectionElementType(), v))
+              .map(v -> toBeamRowFieldValue(Field.of("v", fieldType.getCollectionElementType()), v))
```

Comment:
This worked in an (ad hoc) way that calling methods intended for other uses (Row values) for List values. It also adds some overhead (construct a dumb Field in place) and. Can we add support to null value in toBeamValue instead?

Reply:
This means I should also remove the support for null values from `toBeamRowValue`, since it will be redundant with null support in `toBeamValue`

Reply:
I see, toBeamRowFieldValue is a wrapper of toBeamValue with null check. Then could just rename `toBeamRowFieldValue` to sth like `toBeamMaybeNullValue` and use it in both place?

Reply:
Sounds good, will rename toBeamRowFieldValue.

Reply:
Merge toBeamRowFieldValue and toBeamValue sounds also a good choice. If do so we may need to change toBeamValue's signature to (Field, Object), To retain the ability to get the Field name in the error message here

https://github.com/apache/beam/blob/d658e8d141a9db7c972854ad75a140540af99aa5/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java#L639

Reply:
I like that better, I remember thinking of doing this but seemed like too big of a change 😅

Reply:
made the changes, please let me know if the commits should be more atomic